tests/cover/test_type_lookup.py

```diff
@@ -28,7 +28,7 @@
 # Build a set of all types output by core strategies
 blacklist = [
     'builds', 'iterables', 'permutations', 'random_module', 'randoms',
-    'runner', 'sampled_from', 'streaming',
+    'runner', 'sampled_from', 'streaming', 'fspaths',
```

Comment:
The `pathlib.Path` support certainly won't *hurt*, but won't allow resolution of the windows or posix path types (because those are subtypes of the generic path type).

The *ideal* would be to extract `_windows_fspaths()` and `_unix_fspaths` in the strategies module (without `@defines_strategy`, but returning `st.one_of(s)`), and then

```
if PY3 and pathlib is not None:  # pragma: no branch
     _global_type_lookup.update({
        pathlib.PurePosixPath: st._unix_fspaths().map(os.fsdecode).map(pathlib.PurePosixPath),
        pathlib.PureWindowsPath: st._windows_fspaths().map(os.fsdecode).map(pathlib.PureWindowsPath),
    })
    if os.name == 'nt':  # pragma: no branch
        _global_type_lookup[pathlib.WindowsPath] = st._windows_fspaths().map(os.fsdecode).map(pathlib.WindowsPath)
    else:
        _global_type_lookup[pathlib.PosixPath] = st._unix_fspaths().map(os.fsdecode).map(pathlib.PosixPath)
```

However I can see this is starting to get out of scope, so feel free to leave it for a later enhancement if you prefer!